### PR TITLE
Update agilent drivers to conform to docs standard

### LINKED ISSUE #4365: Typo in the name of the Agilent E8257D driver
I was looking for the driver for our Agilent E8257D, and it took a while to find that the driver is called E8**52**7D in qcodes (flipped 2 and 5). Note that the [unit test](https://github.com/QCoDeS/Qcodes/blob/master/qcodes/tests/drivers/test_Agilent_E8527D.py#L21) where the IDN is checked is correct.

It should be fairly straightforward to rename the class and create a deprecated class with the old name, but if that's too much incompatibility, it would be enough to mention this in the docstring of the driver so that it would show up in a search for "8257D" (that's how I was trying to find the driver).

### PR DESCRIPTION
**Pull request by jenshnielsen**
Updates to implement #4368  for Agilent. Closes #4365 

@mgunyho Could you test the new driver. I have updated the name of the variable and gotten rid of the not needed deg2rad converters and replaced them with numpy

I also suggest that we get rid of the autogenerated docs for the instruments and replace them with something like this.
I would like to add autodocsumm for these modules but am facing the same issue as in https://github.com/QCoDeS/Qcodes/pull/4273

### CODE REVIEW

File: qcodes/instrument_drivers/agilent/E8527D.py [hunk 1]
```diff
@@ -6,8 +6,10 @@
 import qcodes.validators as vals
 from qcodes.instrument import VisaInstrument
 from qcodes.parameters import create_on_off_val_mapping
+from qcodes.utils import deprecate
 
 
+@deprecate(alternative="AgilentE8257D")
 class Agilent_E8527D(VisaInstrument):
```
[the remark this answers is not in the document]
Comment (jenshnielsen) [reply qcodes/instrument_drivers/agilent/E8527D.py:13]: I made backwards incompatible changes in the new driver renaming the status parameter so it seemed easier to just copy it  than creating a baseclass etc.

**Review by astafan8 — APPROVED** [review]

**Comment by jenshnielsen** [comment]
Will postpone merging this until we verify this fixes https://github.com/QCoDeS/Qcodes/issues/4365 and make a decision on https://github.com/QCoDeS/Qcodes/pull/4273 since the docs are hard to navigate without autosummary (due to all the new classes etc)

**Comment by mgunyho** [comment]
I tested with an E8257D unit that the new driver works, I tried `snapshot(update=True)` and setting output_enabled, power, frequency and phase, I checked on the device that they were set.

**Comment by jenshnielsen** [comment]
@mgunyho Thanks 

bors merge

**Pull request merged**